### ISSUE TITLE
3D Object Detection finetune on ScanNet with annotations of 18 object categories

### ISSUE THREAD

**Issue by 3i-cao** [manycore-research/SpatialLM#110]
Hello the authors,
Is checkpoint of finetuned on ScanNet same with checkpoint you provided on huggingface? If not, are there anyway to access the checkpoint finetuned on ScanNet dataset?

**Comment by SamMaoYS** [comment]
Hi, No, these are not the same checkpoints. The results reported on ScanNet come from a different model that was fine-tuned on ScanNet. We do not have plan to release that model, but if your have strong request, I will start working on making it available.

**Comment by 3i-cao** [comment]
hello, thank you for the positive response, since scannet is popular dataset, more information of pretrain on this dataset is really valuable for research community, can you provide the guide to pretrain on ScanNet as you did really well on ARKitScenes? Just pick one scan for train and one for test is enough and useful for further research

**Comment by SamMaoYS** [comment]
In the paper we describe there are some extra steps for tuning on ScanNet: "_For fine-tuning on ScanNet, we introduce an additional pre-training stage to mitigate the difference in label spaces across datasets. Specifically, we map the classes in our dataset to those of ScanNet, and further train the model on our dataset for 3 epochs, followed by fine-tuning on ScanNet for 30 epochs._"
The SpatialLM-Dataset is released, if you have strong interest in reproducing the ScanNet experiment, we recommend building upon the provided training scripts and implementing the necessary modifications for the different training stages. Currently, we do not plan to update the released training code specifically for ScanNet, in order to keep the repository clean and simple. 
If you run into any issues or have detailed questions about the implementation, I’m happy to help.

**Comment by 3i-cao** [comment]
Thank you so much, pls keep this issue for further questions, one again, congrats to your neurips paper, it is really good project

**Comment by SamMaoYS** [comment]
Thank you! As for the checkpoints you requested are provided on huggingface:
- **[ysmao/SpatialLM1.1-Qwen-0.5B-ScanNet-SFT](https://huggingface.co/ysmao/SpatialLM1.1-Qwen-0.5B-ScanNet-SFT)**: Finetuned checkpoint on ScanNet
- **[ysmao/SpatialLM1.1-Qwen-0.5B-Object-Intmd](https://huggingface.co/ysmao/SpatialLM1.1-Qwen-0.5B-Object-Intmd)**: Model pretriained on SpatialLM-Dataset, with labels matched with ScanNet

Let me know if you encountered any issues.

**Comment by bkxmw** [comment]
@SamMaoYS Hello, great work! May I ask if you have any plans to release the processed ScanNet dataset, specifically the one containing pcd (.ply file) and layout (.txt) files?

**Comment by SamMaoYS** [comment]
Hi @bkxmw, thank you. No, currently there's no plan to prepare a "spatiallm-version" of the ScanNet dataset. You can find good reference here in the MMDetection doc [here](https://mmdetection3d.readthedocs.io/en/v1.2.0/advanced_guides/datasets/scannet.html), to preprocess the ScanNet AABB bounding boxes in the textual form.

**Comment by frickyinn** [comment]
@SamMaoYS Hello, thanks a lot for your contributions. Could you kindly provide the hyper-parameters of ScanNet SFT? Such as learning rate, epochs, batch size, etc., I assume the scheduler and optimizer configurations are the same as SFT on ArkitScenes?

**Comment by SamMaoYS** [comment]
Hi @frickyinn, Thank you for your interest. Yes the hyper-parameters remain the same for ScanNet SFT, the learning rate is 1e-4, as described in the paper "we map the classes in our dataset to those of ScanNet, and further train the model on our dataset for 3 epochs, followed by fine-tuning on ScanNet for 30 epochs." On SpatialLM-Dataset the batch size is 64, on ScanNet the batch size is 8. And the random rotation is always disabled.
